pkg/network/protocols/postgres/telemetry_test.go

```diff
@@ -187,8 +194,9 @@ func createEbpfEvent(querySize int) *ebpf.EbpfEvent {
 }
 
 func verifyTelemetry(t *testing.T, tel *Telemetry, expected telemetryResults) {
+
 	for i := 0; i < len(tel.queryLengthBuckets); i++ {
-		assert.Equal(t, expected.queryLength[i], tel.queryLengthBuckets[i].Get(), "queryLength for bucket %d count is incorrect", i)
+		assert.Equal(t, expected.queryLength[i], tel.queryLengthBuckets[i].get(expected.counterState), "queryLength for bucket %d count is incorrect", i)
```

Comment:
After the changes you introduced, the checks are performing different functionality. You are now validating that the metrics you added are correct in that assertion. You need to modify the error message in case of failure, as the current message could be misleading if the test fails.

Reply:
changed